models/utils/int_zendesk__calendar_spine.sql

```diff
@@ -1,11 +1,11 @@
--- depends_on: {{ source('zendesk', 'ticket') }}
+-- depends_on: {{ ref('stg_zendesk__ticket') }}
 
 with spine as (
 
     {% if execute %}
     {% set current_ts = dbt.current_timestamp_backcompat() %}
     {% set first_date_query %}
-        select  min( created_at ) as min_date from {{ source('zendesk', 'ticket') }}
+        select  min( created_at ) as min_date from {{ var('ticket') }}
```

Comment:
Was there a need for these to be changed? If I recall, these were intentionally pointing at the source because if you tried to run `dbt compile` before the staging model was built it will throw an error.

Reply:
yeah this became necessary with my [change](https://github.com/fivetran/dbt_zendesk_source/pull/44/files#diff-3ffdfaf64fda86330c72b8d4a0def7339ab75479dfdab6c57e8cf406cf51848f) to the package-defined zendesk source (ie dynamically disabling it if the customer is using the unioning variables). 

we need to look at the staging model instead, as there's no single source (and the customer isn't _required_ to define sources)

Reply:
I worry that running `dbt compile` before running the package may be a dealbreaker for some users. Is there any possibility for us to allow the use of the source if the user is not leveraging the union feature, and then allow for the switch if they are using the feature?

Reply:
on top of this, would we want to add something like this so union-folks could perhaps compile before running? 
```yml
{% set first_date = run_query(first_date_query).columns[0][0]|string if var('ticket') is not None else '2016-01-01' %}
```

Reply:
eh i suppose this wouldn't really be possible in the pivot model since we're pulling field_names and not just a start_date, so compile would still fail